staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -206,6 +209,32 @@ func compileRule(rule apiextensions.ValidationRule, env *cel.Env, perCallLimit u
 		compilationResult.Error = &apiservercel.Error{apiservercel.ErrorTypeInternal, "cost estimation failed: " + err.Error()}
 		return
 	}
+	if rule.MessageExpression != "" {
+		ast, issues := env.Compile(rule.MessageExpression)
+		if issues != nil {
+			compilationResult.Error = &apiservercel.Error{apiservercel.ErrorTypeInvalid, "messageExpression compilation failed: " + issues.String()}
```

Comment:
Why is the MessageExpression compilation error being used to populate the compliationResult.Error for the main expression?

Reply:
Yeah, I think that's confusing. I've split `MessageExpression`-related errors out into a separate field.